test/adaptive_avg_pool/test_adaptive_avg_pool_nd.py

```diff
@@ -0,0 +1,33 @@
+"""Test the shape checker of AdaptiveAvgPoolNDDerivatives."""
+from test.adaptive_avg_pool.problem import AdaptiveAvgPoolProblem, make_test_problems
+from test.adaptive_avg_pool.settings_adaptive_avg_pool_nd import SETTINGS
+from typing import List
+
+import pytest
+
+PROBLEMS: List[AdaptiveAvgPoolProblem] = make_test_problems(SETTINGS)
+IDS: List[str] = [problem.make_id() for problem in PROBLEMS]
+
+
+@pytest.mark.parametrize("problem", PROBLEMS, ids=IDS)
+def test_adaptive_avg_pool_check_parameters(problem: AdaptiveAvgPoolProblem):
+    """Test AdaptiveAvgPoolNDDerivatives.check_parameters().
+
+    Additionally check if returned parameters are indeed equivalent.
+
+    Args:
+        problem: test problem
+    """
+    problem.set_up()
+    if problem.works:
+        problem.check_parameters()
+        problem.check_equivalence()
+    else:
+        with pytest.raises(NotImplementedError):
+            problem.check_parameters()
+        # TODO improve equivalence coverage and activate these lines
+        # check for perfect coverage:
+        # all cases that have equivalent parameters are recognized
+        # with pytest.raises(AssertionError):
+        #    problem.check_equivalence()
```

Comment:
Is this still a TODO?

Reply:
Yes. Depending on perspective this is either a minor or a major issue.
Probably, I need to improve the explanation, but I'm not sure how to state it concisely.

If `check_parameters` decides there is equivalence, then the equivalence works. (core functionality)
If `check_parameters` decides there is no equivalence, then there are still some cases where the equivalence works.

It would be nice, if all cases that have equivalence are recognized by `check_parameters`. If this was the case, we could activate these lines and the test would still pass (which it doesn't at the moment).